basic-javascript/project-1.spec.js

```diff
@@ -1,3 +1,150 @@
+});
+
+it('should return whether the value is odd', () => {
+    expect(helpers.isOdd(5)).toBe(true);
+    expect(helpers.isOdd(2)).toBe(false);
+    expect(helpers.isOdd(24)).toBe(false);
+});
+
+it('should return the square of the value', () => {
+    expect(helpers.square(5)).toBe(25);
+    expect(helpers.square(12)).toBe(144);
+    expect(helpers.square(3)).toBe(9);
+});
+
+it('should return the cube of the value', () => {
+    expect(helpers.cube(3)).toBe(27);
+    expect(helpers.cube(4)).toBe(64);
+    expect(helpers.cube(5)).toBe(125);
+});
+
+it('should return Math.pow(num, exponent)', () => {
+    expect(helpers.raiseToPower(3, 2)).toBe(9);
+    expect(helpers.raiseToPower(2, 4)).toBe(16);
+    expect(helpers.raiseToPower(5, 3)).toBe(125);
+});
+
+it('should return the rounded number based on rounding rules', () => {
+    expect(helpers.roundNumber(1.2)).toBe(1);
+    expect(helpers.roundNumber(5.45)).toBe(5);
+    expect(helpers.roundNumber(7.7)).toBe(8);
+});
+
+it('should return the number rounded up', () => {
+    expect(helpers.roundUp(1.2)).toBe(2);
+    expect(helpers.roundUp(5.45)).toBe(6);
+    expect(helpers.roundUp(7.7)).toBe(8);
+});
+
+it('should return a string with an exclamation point added', () => {
+    expect(helpers.addExclamationPoint('test')).toBe('test!');
+    expect(helpers.addExclamationPoint('hello, world')).toBe('hello, world!');
+    expect(helpers.addExclamationPoint('tonight')).toBe('tonight!');
+});
+
+it('should return first and last name combined', () => {
+    expect(helpers.combineNames('Grant', 'Reighard')).toBe('Grant Reighard');
+    expect(helpers.combineNames('Ashwin', 'Sundararajan')).toBe('Ashwin Sundararajan');
+    expect(helpers.combineNames('John', 'O\'Rourke')).toBe('John O\'Rourke');
+});
+
+it('should return \'Hello name!\'', () => {
+    expect(helpers.getGreeting('Grant')).toBe('Hello Grant!');
+    expect(helpers.getGreeting('Ashwin')).toBe('Hello Ashwin!');
+    expect(helpers.getGreeting('John')).toBe('Hello John!');
+});
+
+it('should return length times width', () => {
+    expect(helpers.getRectangleArea(10, 20)).toBe(200);
+    expect(helpers.getRectangleArea(5, 100)).toBe(500);
+    expect(helpers.getRectangleArea(30, 20)).toBe(600);
+});
+
+it('should return half of base times height', () => {
+    expect(helpers.getTriangleArea(5, 10)).toBe(25);
+    expect(helpers.getTriangleArea(10, 10)).toBe(50);
+    expect(helpers.getTriangleArea(20, 13)).toBe(130);
+});
+
+it('should return Pi times radius squared', () => {
+    expect(helpers.getCircleArea(5)).toBeCloseTo(78.54);
+    expect(helpers.getCircleArea(100)).toBeCloseTo(31415.93);
+    expect(helpers.getCircleArea(20)).toBeCloseTo(1256.637);
+});
+
+it('should return length times width times height', () => {
+    expect(helpers.getRectangularPrismVolume(10, 10, 10)).toBe(1000);
+    expect(helpers.getRectangularPrismVolume(5, 10, 25)).toBe(1250);
+    expect(helpers.getRectangularPrismVolume(3, 4, 5)).toBe(60);
+});
```

Comment:
Your test looks good but you should think about testing edge cases, like when a user put string instead of a number. you test should handle those.